src/components/forms/ComboBox/ComboBox.tsx

```diff
@@ -78,372 +85,391 @@ const Input = ({
   )
 }
 
-export const ComboBox = ({
-  id,
-  name,
-  className,
-  options,
-  defaultValue,
-  disabled,
-  onChange,
-  assistiveHint,
-  noResults,
-  selectProps,
-  inputProps,
-  ulProps,
-  customFilter,
-  disableFiltering = false,
-}: ComboBoxProps): React.ReactElement => {
-  const isDisabled = !!disabled
-
-  let defaultOption
-  if (defaultValue) {
-    defaultOption = options.find((opt: ComboBoxOption): boolean => {
-      return opt.value === defaultValue
-    })
-  }
-
-  const filter: CustomizableFilter = customFilter
-    ? customFilter
-    : { filter: DEFAULT_FILTER }
-
-  const initialState: State = {
-    isOpen: false,
-    selectedOption: defaultOption ? defaultOption : undefined,
-    focusedOption: undefined,
-    focusMode: FocusMode.None,
-    filteredOptions: options,
-    inputValue: defaultOption ? defaultOption.label : '',
-  }
-
-  const [state, dispatch] = useComboBox(
-    initialState,
-    options,
-    disableFiltering,
-    filter
-  )
-
-  const containerRef = useRef<HTMLDivElement>(null)
-  const focusedItemRef = useRef<HTMLLIElement>(null)
-
-  useEffect(() => {
-    onChange && onChange(state.selectedOption?.value || undefined)
-  }, [state.selectedOption])
+export type ComboBoxRef = {
+  clearSelection: () => void
+}
```

Comment:
Awesome! 

Do you foresee the need to expose access to any of the child elements of combo box, similar to how the `input` for FileInput is exposed? I don't think that would be a breaking change if added in the future, so I'd be okay with a github issue for doing that as well.

Reply:
I can't really foresee a need to expose the internal child elements? There's nothing that complicated going on under the hood, and the `clearSelection` method is just a way to make the component partially controlled. Curious what the use case might be for that.

But yeah, it's just adding the `forwardRef` that's the breaking change; additional functionality on the ref isn't breaking. I'm generally of the YAGNI opinion here; if someone asks for access to the child elements, we can just do it then.

Reply:
Works for me!